cloud/docker/docker_login.py

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/python
+#
+# (c) 2016 Olaf Kilian <olaf.kilian@symanex.com>
+#          Chris Houseknecht, <house@redhat.com>
+#          James Tanner, <jtanner@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+DOCUMENTATION = '''
+---
+module: docker_login
+
+short_description: Log into a Docker registry.
+
+version_added: "2.0"
+
+description:
+    - Provides functionality similar to the "docker login" command.
+    - Authenticate with a docker registry and add the credentials to your local Docker config file. Adding the
+      credentials to the config files allows future connections to the registry using tools such as Ansible's Docker
+      modules, the Docker CLI and docker-py without needing to provide credentials.
+    - Running in check mode will perform the authentication without updating the config file.
+
+options:
```

Comment:
Not sure what you're after. In the origin module docker_url was defined as:

```
docker_url:
    description:
       - Refers to the protocol+hostname+port where the Docker server is hosted
    required: false
    default: unix://var/run/docker.sock
```

In the new Docker modules the equivalent is `docker_host` (with an alias of 'docker_url') which is defined in docker_common.py